src/fs/rock/RockIoState.cc

```diff
@@ -115,6 +118,13 @@ Rock::IoState::read_(char *buf, size_t len, off_t coreOff, STRCB *cb, void *data
     read.callback = cb;
     read.callback_data = cbdataReference(data);
 
+    // quit if we cannot read what they want, and the writer cannot add more
+    if (sidCurrent < 0 && writerLeft) {
```

Comment:
We can probably combine this if and the if below:

```suggestion
    if (sidCurrent < 0) {
        if (writeLeft) 
        [...]
    }
```

Reply:
Agreed, but I prefer the consistency of the current "one if, one return" code structure. It might help with understanding each terminal condition better, and it reduces code changes. This is a slight preference -- both variants would be OK.